src_c/event.c

```diff
@@ -1901,6 +1919,29 @@ _pg_get_all_events(void)
         for (loop = 0; loop < len; loop++) {
             if (!_pg_event_append_to_list(list, &eventbuf[loop]))
                 goto error;
+
+            if (eventbuf[loop].type == SDL_KEYDOWN) {
+                const Uint8 *keyboard_state = SDL_GetKeyboardState(NULL);
+                int i;
+                for (i = 0; i < NUM_KEYS; i++) {
+                    if (keyboard_state[i] && !keys_being_held_down[i]) {
+                        held_down_keys[i] = 1;
+                        keys_being_held_down[i] = 1;
+                    }
+                }
+            }
```

Comment:
This is not what I had in mind. Now I understand why you documented it like that. You should use the KEYDOWN or KEYUP event itself, use it's data, not poll the whole keyboard every event.

Reply:
Okay thanks, that makes a lot more sense than what i was doing